File: mobile-client/ios/UseCase/Auth/Session/RegisterAuthUseCase.swift
public struct RegisterAuthUseCase {
    private let gateway: AuthSessionGateway

    public init(gateway: AuthSessionGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(request: RegisterAuthRequest) async throws -> AuthSessionPayload {
        try await gateway.registerAuth(request: request)
    }
}

File: mobile-client/android/UseCase/Auth/Session/RegisterAuthUseCase.kt
package app.mobiling.client.usecase.auth.session

import app.mobiling.client.contract.auth.session.AuthSessionPayload
import app.mobiling.client.contract.auth.session.RegisterAuthRequest
import app.mobiling.client.data.auth.session.AuthSessionGateway

class RegisterAuthUseCase(
    private val gateway: AuthSessionGateway,
) {
    suspend operator fun invoke(request: RegisterAuthRequest): AuthSessionPayload = gateway.registerAuth(request)
}
